pkg/querier/queryrange/results_cache.go

```diff
@@ -195,7 +237,19 @@ func (s resultsCache) handleMiss(ctx context.Context, r Request) (Response, []Ex
 		return response, []Extent{}, nil
 	}
 
-	extent, err := toExtent(ctx, r, response)
+	if s.cacheGenNumberLoader != nil {
```

Comment:
We shouldn't be doing it at this level imo. We also cache in `handleHit` case. Can we move this logic into `Do()`?